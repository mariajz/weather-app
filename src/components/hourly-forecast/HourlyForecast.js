import React from 'react';
import { ScrollView } from 'react-native';
import { StyledText } from '../../commons/styles';
import { CustomIcon } from '../../commons/visual-elements';
import useCurrentHour from '../../states/useCurrentHour';
import { filterForecastDataFromCurrentHour } from '../../utils/helpers';
import {
    CalenderWrapper,
    HorizontalScrollWrapper,
    HourlyForecastWrapper,
} from './HourlyForecast.style';
import HourlySummary from './HourlySummary';

const HourlyForecast = ({ hourlyForecastProps }) => {
    const { forecastData } = hourlyForecastProps;
    const { currentHour } = useCurrentHour();

    const filteredForecastData = filterForecastDataFromCurrentHour(
        forecastData,
        currentHour,
    );

    return (
        <HourlyForecastWrapper>
            <CalenderWrapper>
                <CustomIcon iconName="calendar" />
                <StyledText marginLeft={5} size={12}>
                    HOURLY FORECAST
                </StyledText>
            </CalenderWrapper>
            <HorizontalScrollWrapper>
                <ScrollView horizontal showsHorizontalScrollIndicator={false}>
                    {filteredForecastData.map(item => (
                        <HourlySummary
                            hourlyWeatherItem={item}
                            key={item.time}
                        />
                    ))}
                </ScrollView>
            </HorizontalScrollWrapper>
        </HourlyForecastWrapper>
    );
};

export default HourlyForecast;
