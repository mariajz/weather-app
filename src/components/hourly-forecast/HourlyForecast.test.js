import { render } from '@testing-library/react-native';
import React from 'react';
import HourlyForecast from './HourlyForecast';

const defaultProps = {
    hourlyForecastProps: {
        forecastData: [
            {
                condition: 'Cloudy',
                temp: 21.1,
                time: '7 AM',
                twentyFourHourFormat: '2023-08-14 07:00',
                weatherImage: 13,
            },
            {
                condition: 'Sunny',
                temp: 22.2,
                time: '8 AM',
                twentyFourHourFormat: '2023-08-14 08:00',
                weatherImage: 12,
            },
            {
                condition: 'Overcast',
                temp: 25.8,
                time: '9 AM',
                twentyFourHourFormat: '2023-08-14 09:00',
                weatherImage: 13,
            },
        ],
    },
};

jest.mock('../../states/useCurrentHour', () => () => ({
    currentHour: 8,
}));

describe('HourlyForecast', () => {
    it('should render HourlyForecast', () => {
        const container = render(<HourlyForecast {...defaultProps} />);

        expect(container).toMatchSnapshot();
    });
});
