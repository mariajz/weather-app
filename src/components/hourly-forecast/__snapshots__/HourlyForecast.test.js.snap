// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HourlyForecast should render HourlyForecast 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(255,255,255, 0.2)",
      "borderBottomLeftRadius": 10,
      "borderBottomRightRadius": 10,
      "borderTopLeftRadius": 10,
      "borderTopRightRadius": 10,
      "marginBottom": 15,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginBottom": 10,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../assets/icons/calendar.png",
        }
      }
      style={
        {
          "height": 25,
          "width": 25,
        }
      }
      testID="icon"
      tintColor="white"
    />
    <Text
      marginLeft={5}
      size={12}
      style={
        {
          "color": "white",
          "fontSize": 12,
          "marginLeft": 5,
        }
      }
    >
      HOURLY FORECAST
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
        "width": "100%",
      }
    }
  >
    <RCTScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "column",
              "height": 60,
              "justifyContent": "space-between",
              "maxWidth": 60,
              "paddingBottom": 0,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 0,
            }
          }
        >
          <Text
            size={10}
            style={
              {
                "color": "white",
                "fontSize": 10,
                "marginLeft": 0,
              }
            }
          >
            8 AM
          </Text>
          <Image
            height={25}
            source={12}
            style={
              {
                "height": 25,
                "width": 25,
              }
            }
            testID="image"
            width={25}
          />
          <Text
            size={10}
            style={
              {
                "color": "white",
                "fontSize": 10,
                "marginLeft": 0,
              }
            }
          >
            22.2
            °
            C
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "column",
              "height": 60,
              "justifyContent": "space-between",
              "maxWidth": 60,
              "paddingBottom": 0,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 0,
            }
          }
        >
          <Text
            size={10}
            style={
              {
                "color": "white",
                "fontSize": 10,
                "marginLeft": 0,
              }
            }
          >
            9 AM
          </Text>
          <Image
            height={25}
            source={13}
            style={
              {
                "height": 25,
                "width": 25,
              }
            }
            testID="image"
            width={25}
          />
          <Text
            size={10}
            style={
              {
                "color": "white",
                "fontSize": 10,
                "marginLeft": 0,
              }
            }
          >
            25.8
            °
            C
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
